src/conf/domain_conf.c

```diff
@@ -19897,19 +19901,69 @@ virDomainDefParseXML(xmlXPathContextPtr ctxt,
                                                        ctxt,
                                                        nodes[i],
                                                        flags);
+
         if (!chr)
             return NULL;
 
-        if (chr->target.port == -1) {
+        def->serials[def->nserials++] = chr;
```

Comment:
follow libvirt coding guidelines . Might need a tab

Reply:
not needed. Checked